packages/@aws-cdk/assertions/lib/template.ts

```diff
@@ -109,46 +109,50 @@ export class Template {
    * Assert that an Output with the given properties exists in the CloudFormation template.
    * By default, performs partial matching on the resource, via the `Match.objectLike()`.
    * To configure different behavour, use other matchers in the `Match` class.
+   * @param outputName the name of the output. Provide '*' to match all Output names in the template.
    * @param props the output as should be expected in the template.
    */
-  public hasOutput(props: any): void {
-    const matchError = hasOutput(this.inspector, props);
+  public hasOutput(outputName: string, props: any): void {
```

Comment:
In Mappings they call it "logical name" - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/mappings-section-structure.html, and in Outputs they call it "logical id" - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html.
I suggest we use "logical id" in both cases.

Let's use `logicalId` everywhere instead of `outputName`.